service/tablet_allocator.cc

```diff
@@ -2049,11 +2050,29 @@ class load_balancer {
 
         auto get_viable_targets = [&] () {
             std::unordered_set<host_id> viable_targets;
+            bool same_rack = false;
```

Comment:
Nit: the name `same_rack` implies as this bool meant that some things are on the same rack, whereas this expresses that tablets _must be_ on the same rack. How about `must_be_on_the_same_rack`?

Reply:
Makes sense, added in the rebase